internal/ir/infra.go

```diff
@@ -34,6 +35,11 @@ func (i Infra) YAMLString() string {
 	return string(y)
 }
 
+func (i Infra) JSONString() string {
```

Comment:
Struct Infra has methods on both value and pointer receivers. Such usage is not recommended by the Go Documentation.